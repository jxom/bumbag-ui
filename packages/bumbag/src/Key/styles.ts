import { cssClass } from '../styled';
import { borderRadius, palette, theme } from '../utils';

export const Key = (styleProps) => cssClass`
  box-sizing: border-box;
  display: inline;
  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
  background-color: ${palette('default')(styleProps)};
  font-size: 0.875em;
  border-radius: ${borderRadius('2')(styleProps)};
  padding: 0.25em 0.5em;
  border-style: solid;
  border-color: ${palette('white900')(styleProps)};
  border-width: 1px 1px 2px;

  & {
    ${theme(styleProps.themeKey, `css.root`)(styleProps)};
  }
`;
