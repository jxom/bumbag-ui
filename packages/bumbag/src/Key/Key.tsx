import { Box as ReakitBox } from 'reakit';

import { useClassName, createComponent, createElement, createHook } from '../utils';
import { Box, BoxProps } from '../Box';

import * as styles from './styles';

export type LocalKeyProps = {};
export type KeyProps = BoxProps & LocalKeyProps;

const useProps = createHook<KeyProps>(
  (props, { themeKey, themeKeyOverride }) => {
    const boxProps = Box.useProps(props);

    const className = useClassName({
      style: styles.Key,
      styleProps: props,
      themeKey,
      themeKeyOverride,
      prevClassName: boxProps.className,
    });

    return { ...boxProps, className };
  },
  { themeKey: 'Key' }
);

export const Key = createComponent<KeyProps>(
  (props) => {
    const textProps = useProps(props);
    return createElement({ children: props.children, component: ReakitBox, use: props.use, htmlProps: textProps });
  },
  {
    attach: {
      useProps,
    },
    defaultProps: {
      use: 'kbd',
    },
    themeKey: 'Key',
  }
);
