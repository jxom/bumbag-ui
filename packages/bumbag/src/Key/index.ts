import * as styles from './styles';

export * from './Key';
export { styles as keyStyles };
