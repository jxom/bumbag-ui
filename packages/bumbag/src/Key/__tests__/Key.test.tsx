import * as React from 'react';
import { renderHook } from '@testing-library/react-hooks';
import { Key } from '../Key';
import render from '../../utils/_tests/render';

describe('props', () => {
  it('should render correctly', () => {
    const { container } = render(<Key>Space</Key>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with CSS props', () => {
    const { container } = render(<Key color="primary">Space</Key>);
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('variants', () => {
  it('css.root should render correctly', () => {
    const { container } = render(<Key variant="test">Space</Key>, {
      theme: { Key: { variants: { test: { css: { root: { backgroundColor: 'red' } } } } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('overrides', () => {
  it('css.root should render correctly', () => {
    const { container } = render(<Key overrides={{ Key: { css: { root: { backgroundColor: 'red' } } } }}>Space</Key>);
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('theming', () => {
  it('css.root should render correctly', () => {
    const { container } = render(<Key>Space</Key>, {
      theme: { Key: { css: { root: { backgroundColor: 'red' } } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('defaultProps', () => {
  it('should render correctly for className', () => {
    const { container } = render(<Key>Space</Key>, {
      theme: { Key: { defaultProps: { className: 'test', color: 'primary' } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
