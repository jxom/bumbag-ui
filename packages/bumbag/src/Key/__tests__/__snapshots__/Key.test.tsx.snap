// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`defaultProps should render correctly for className 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  box-sizing: border-box;
  display: inline;
  -webkit-tap-highlight-color: rgba(0,0,0,0);
  background-color: var(--bb-palette-default,#000000);
  font-size: 0.875em;
  border-radius: 4px;
  padding: 0.25em 0.5em;
  border-style: solid;
  border-color: var(--bb-palette-white900,#000000);
  border-width: 1px 1px 2px;
}

.emotion-2 {
  color: var(--bb-palette-primary,#000000) !important;
}

<kbd
  class="emotion-0 emotion-1 emotion-2 test bb-Box bb-Key"
>
  Space
</kbd>
`;

exports[`overrides styles.base should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  box-sizing: border-box;
  display: inline;
  -webkit-tap-highlight-color: rgba(0,0,0,0);
  background-color: var(--bb-palette-default,#000000);
  font-size: 0.875em;
  border-radius: 4px;
  padding: 0.25em 0.5em;
  border-style: solid;
  border-color: var(--bb-palette-white900,#000000);
  border-width: 1px 1px 2px;
}

.emotion-0 {
  background-color: var(--bb-palette-red,#ff0000);
}

<kbd
  class="emotion-0 emotion-1 emotion-2 bb-Box bb-Key"
>
  Space
</kbd>
`;

exports[`props should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  box-sizing: border-box;
  display: inline;
  -webkit-tap-highlight-color: rgba(0,0,0,0);
  background-color: var(--bb-palette-default,#000000);
  font-size: 0.875em;
  border-radius: 4px;
  padding: 0.25em 0.5em;
  border-style: solid;
  border-color: var(--bb-palette-white900,#000000);
  border-width: 1px 1px 2px;
}

<kbd
  class="emotion-0 emotion-1 emotion-2 bb-Box bb-Key"
>
  Space
</kbd>
`;

exports[`props should render correctly with CSS props 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  box-sizing: border-box;
  display: inline;
  -webkit-tap-highlight-color: rgba(0,0,0,0);
  background-color: var(--bb-palette-default,#000000);
  font-size: 0.875em;
  border-radius: 4px;
  padding: 0.25em 0.5em;
  border-style: solid;
  border-color: var(--bb-palette-white900,#000000);
  border-width: 1px 1px 2px;
}

.emotion-2 {
  color: var(--bb-palette-primary,#000000) !important;
}

<kbd
  class="emotion-0 emotion-1 emotion-2 bb-Box bb-Key"
>
  Space
</kbd>
`;

exports[`theming styles.base should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  box-sizing: border-box;
  display: inline;
  -webkit-tap-highlight-color: rgba(0,0,0,0);
  background-color: var(--bb-palette-default,#000000);
  font-size: 0.875em;
  border-radius: 4px;
  padding: 0.25em 0.5em;
  border-style: solid;
  border-color: var(--bb-palette-white900,#000000);
  border-width: 1px 1px 2px;
}

.emotion-0 {
  background-color: var(--bb-palette-red,#ff0000);
}

<kbd
  class="emotion-0 emotion-1 emotion-2 bb-Box bb-Key"
>
  Space
</kbd>
`;

exports[`variants styles.base should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  box-sizing: border-box;
  display: inline;
  -webkit-tap-highlight-color: rgba(0,0,0,0);
  background-color: var(--bb-palette-default,#000000);
  font-size: 0.875em;
  border-radius: 4px;
  padding: 0.25em 0.5em;
  border-style: solid;
  border-color: var(--bb-palette-white900,#000000);
  border-width: 1px 1px 2px;
}

.emotion-0 {
  background-color: var(--bb-palette-red,#ff0000);
}

<kbd
  class="emotion-0 emotion-1 emotion-2 bb-Box bb-Key"
>
  Space
</kbd>
`;
